packages/gatsby-transformer-screenshot/src/gatsby-node.js

```diff
@@ -76,6 +76,14 @@ exports.onPreBootstrap = (
   })
 }
 
+exports.sourceNodes = ({ actions: { createTypes } }) => {
+  createTypes(/* GraphQL */ `
+    type Screenshot implements Node {
```

Comment:
I feel like we should have `@childOf(types: ["SitesYaml"])` here to automatically create `childScreenshot` field on parent type. Just one thing here is that we allow other node types (via `pluginOptions.nodeTypes`) - so we would have to construct list of types dynamically and then SDL becomes iffy so maybe swapping this to object builder would be the way to go?

```js
exports.schemaCustomization = ({ actions: { createTypes }, schema }, pluginOptions) => {
  const validNodeTypes = [`SitesYaml`].concat(pluginOptions.nodeTypes || [])

  createTypes(schema.buildObjectType({
    name: `Screenshot`,
    fields: {
      screenshotFile: {
        type: `File`
        extensions: {
          link: {}
        }
      }
    },
    interfaces: [`Node`],
    extensions: {
      childOf: {
        types: validNodeTypes
      }
    }
  })
})
```

Reply:
That sounds good. Would we need to extend the nodes defined in `validNodeTypes` to add `childScreenshot`? Or does adding `childOf` make that automatic?

Reply:
Also, since this is adding a fair amount of logic, what's the best way to test this?